profile/filter_test.go

```diff
@@ -0,0 +1,252 @@
+		{ID: 2, Start: 0x50000, Limit: 0x70000, File: "map1", HasFunctions: true, HasFilenames: true, HasLineNumbers: true, HasInlineFrames: true},
+	}
+	functions := []*Function{
+		{ID: 1, Name: "fun0", SystemName: "fun0", Filename: "file0"},
+		{ID: 2, Name: "fun1", SystemName: "fun1", Filename: "file1"},
+		{ID: 3, Name: "fun2", SystemName: "fun2", Filename: "file2"},
+		{ID: 4, Name: "fun3", SystemName: "fun3", Filename: "file3"},
+		{ID: 5, Name: "fun4", SystemName: "fun4", Filename: "file4"},
+		{ID: 6, Name: "fun5", SystemName: "fun5", Filename: "file5"},
+		{ID: 7, Name: "fun6", SystemName: "fun6", Filename: "file6"},
+		{ID: 8, Name: "fun7", SystemName: "fun7", Filename: "file7"},
+		{ID: 9, Name: "fun8", SystemName: "fun8", Filename: "file8"},
+		{ID: 10, Name: "fun9", SystemName: "fun9", Filename: "file9"},
+		{ID: 11, Name: "fun10", SystemName: "fun10", Filename: "file10"},
+	}
+	locations := []*Location{
+		{ID: 1, Mapping: mappings[0], Address: 0x1000, Line: []Line{{Function: functions[0], Line: 1}}},
+		{ID: 2, Mapping: mappings[0], Address: 0x2000, Line: []Line{{Function: functions[1], Line: 1}}},
+		{ID: 3, Mapping: mappings[0], Address: 0x3000, Line: []Line{{Function: functions[2], Line: 1}}},
+		{ID: 4, Mapping: mappings[0], Address: 0x4000, Line: []Line{{Function: functions[3], Line: 1}}},
+		{ID: 5, Mapping: mappings[0], Address: 0x5000, Line: []Line{{Function: functions[4], Line: 1}}},
+		{ID: 6, Mapping: mappings[0], Address: 0x6000, Line: []Line{{Function: functions[5], Line: 1}}},
+		{ID: 7, Mapping: mappings[0], Address: 0x7000, Line: []Line{{Function: functions[6], Line: 1}}},
+		{ID: 8, Mapping: mappings[0], Address: 0x8000, Line: []Line{{Function: functions[7], Line: 1}}},
+		{ID: 9, Mapping: mappings[0], Address: 0x9000, Line: []Line{{Function: functions[8], Line: 1}}},
+		{ID: 10, Mapping: mappings[0], Address: 0x10000, Line: []Line{{Function: functions[9], Line: 1}}},
+		{ID: 11, Mapping: mappings[1], Address: 0x11000, Line: []Line{{Function: functions[10], Line: 1}}},
+	}
+	profile := &Profile{
+		TimeNanos:     10000,
+		PeriodType:    &ValueType{Type: "cpu", Unit: "milliseconds"},
+		Period:        1,
+		DurationNanos: 10e9,
+		SampleType: []*ValueType{{Type: "samples", Unit: "count"}},
+		Mapping:  mappings,
+		Function: functions,
+		Location: locations,
+		Sample: []*Sample{
+			{Value: []int64{1}, Location: []*Location{locations[0], locations[1], locations[2], locations[3]}},
+			{Value: []int64{2}, Location: []*Location{locations[4], locations[5], locations[1], locations[6]}},
+			{Value: []int64{3}, Location: []*Location{locations[7], locations[8]}},
+			{Value: []int64{4}, Location: []*Location{locations[9], locations[4], locations[10], locations[7]}},
+		},
+	}
+
+	for _, testCase := range []struct{
+		name string
+		focus, ignore, hide, show *regexp.Regexp
+		// todo: fm, im, hm, hnm           bool
+		wantPaths [][]int // expected paths. wantPaths[sampleIndex][frameIndex] = locationIndex
```

Comment:
Nit: proper sentences in comments please.

Reply:
done